pyccel/ast/numpyext.py

```diff
@@ -1208,15 +1223,47 @@ def __init__(self, x):
         self._dtype     = NativeReal()
         self._precision = default_precision[str_dtype(self._dtype)]
 
+class NumpyMod(NumpyUfuncBinary):
+    def __init__(self, *args):
+        integers  = [a for a in args if a.dtype is NativeInteger() or a.dtype is NativeBool()]
+        reals     = [a for a in args if a.dtype is NativeReal()]
+        others    = [a for a in args if a not in integers+reals]
+
+        if others:
+            raise TypeError('{} not supported'.format(others[0].dtype))
+
+        if reals:
+            self._dtype     = NativeReal()
+            self._precision = max(a.precision for a in reals)
+        elif integers:
+            self._dtype     = NativeInteger()
+            self._precision = max(a.precision for a in integers)
+        else:
+            raise TypeError('cannot determine the type of {}'.format(self))
+
+        shapes = [a.shape for a in args]
 
-class Min(Function, PyccelAstNode):
-    def _eval_is_integer(self):
-        return all(i.is_integer for i in self.args)
+        if all(sh is not None for sh in shapes):
+            if len(args) == 1:
+                shape = args[0].shape
+            else:
+                shape = broadcast(args[0].shape, args[1].shape)
 
-class Max(Function, PyccelAstNode):
-    def _eval_is_integer(self):
-        return all(i.is_integer for i in self.args)
+                for a in args[2:]:
+                    shape = broadcast(shape, a.shape)
 
+            self._shape = shape
+            self._rank  = len(shape)
+        else:
+            self._rank = max(a.rank for a in args)
+
+class NumpyMin(NumpyUfuncUnary):
+    def __init__(self, x):
+        self.copy_attributes(x)
```

Comment:
Doesn't this mean that `np.min(x)` has the same shape and rank as x?

Reply:
that's correct

Reply:
Which numpy function does this represent?
`http://pageperso.lif.univ-mrs.fr/~francois.denis/IAAM1/numpy-html-1.14.0/reference/generated/numpy.amin.html#numpy.amin` ? (wrong name)
`https://numpy.org/doc/stable/reference/generated/numpy.minimum.html` (wrong number of arguments)
`http://pageperso.lif.univ-mrs.fr/~francois.denis/IAAM1/numpy-html-1.14.0/reference/generated/numpy.ndarray.min.html` (pyccel doesn't handle `a.min()` syntax

Reply:
you can do `from numpy import min`

Reply:
Ok, that's what I assumed. You are right it seems to be equivalent to amin. It's odd that I can't find documentation for it though